src/crypto/equihash.cpp

```diff
@@ -16,6 +16,10 @@
 #include "config/bitcoin-config.h"
 #endif
 
+#if defined(__APPLE__)
+#include "compat/endian.h"
+#endif
```

Comment:
Yes, it looks like ``be32toh`` is getting accidentally included by other headers on Linux, so explicitly including ``compat/endian.h`` is better.